tools/mirror/src/lib.rs

```diff
@@ -212,18 +208,74 @@ fn delete_pending_outcome(db: &DB, id: &CryptoHash) -> anyhow::Result<()> {
     )?)
 }
 
-fn set_last_source_height(db: &DB, height: BlockHeight) -> anyhow::Result<()> {
-    // TODO: we should instead save something like the
-    // (block_height, shard_id, idx_in_chunk) of the last
-    // transaction sent. Currently we set last_source_height after
-    // sending all of the transactions in that chunk, so if we get
+#[derive(Debug, BorshDeserialize, BorshSerialize)]
+enum TxBookmark {
```

Comment:
can you add comments to the structs and fields about what they represent and/or how they are used?